### PR TITLE
[Experimental] Use CheckboxList component in interactivity attribute filter

### LINKED ISSUE #43171: Filter by attribute product count in parens should not have spaces
Image

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

To solve https://github.com/woocommerce/woocommerce/issues/43171 the simplest thing was to do the work we needed to do anyway: move CollectionAttributeFilter to use the reusable CheckboxList component.

Closes #43171

### How to test the changes in this Pull Request:

1. Add attribute filter to page or template
2. Test it as normal in checkbox mode, ensuring that selecting options still filters down the products as before

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
[Experimental] Use CheckboxList component in interactivity attribute filter

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php [hunk 1]
```diff
@@ -2,7 +2,7 @@
 namespace Automattic\WooCommerce\Blocks\BlockTypes;
 
 use Automattic\WooCommerce\Blocks\InteractivityComponents\Dropdown;
-use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
```
Comment (samueljseay) [inline plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php:5]: These includes were accidentally left in by me when I refactored Dropdown

File: plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php [hunk 2]
```diff
@@ -179,7 +179,7 @@ function( $term ) use ( $attribute_counts, $selected_terms ) {
 			$attribute_terms
 		);
 
-		$filter_content = 'dropdown' === $attributes['displayStyle'] ? $this->render_attribute_dropdown( $attribute_options, $attributes ) : $this->render_attribute_list( $attribute_options, $attributes );
+		$filter_content = 'dropdown' === $attributes['displayStyle'] ? $this->render_attribute_dropdown( $attribute_options, $attributes ) : $this->render_attribute_checkbox_list( $attribute_options, $attributes );
```
Comment (dinhtungdu) [inline plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php]: We're passing `$attributes` array but used it as `$show_counts` boolean value in the `render_attribute_checkbox_list` method body.

**Review by dinhtungdu — COMMENTED** [review]
This is definitely the right move. Apart from a review comment, this is LGTM!

File: plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php (hunk 2 above)
Comment (samueljseay) [reply plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php]: whoops! thanks for spotting

Comment (samueljseay) [reply plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php]: It felt inconsistent to have both methods take different sigs so for consistency they both take attributes now and we determine from `$attributes['showCounts']` whether to show counts. https://github.com/woocommerce/woocommerce/pull/43217/commits/223c796e2f9fd164084a4610906bc6bf224eb486

**Review by dinhtungdu — APPROVED** [review]

**Pull request merged**